ruby/red-arrow/test/test-slicer.rb

```diff
@@ -528,5 +540,40 @@ def setup
 2	window
       TABLE
     end
+
+    test("match_like") do
+      sliced_table = @table.slice do |slicer|
+        slicer.string.match_like("_rr%")
+      end
+      assert_equal(<<~TABLE, sliced_table.to_s)
+	string
+0	array 
+1	Arrow 
+2	(null)
+      TABLE
+    end
+
+    test("match_substring_regex") do
+      sliced_table = @table.slice do |slicer|
+        slicer.string.match_substring_regex("[dr]ow")
```

Comment:
Can we accept `/[dr]ow/` too?

If `slicer.string.match_substring_regex(/[dr]ow/)` works, how about adding support for `slicer.string.match_substring(/[dr]ow/)` (== `slicer.string.amtch_substring_regex(/[dr]ow/)`)`?

Reply:
Yes we can do that!

Reply:
Implemented above.

And renamed `match_substring_regex?` to `match_substring_regexp?`.

Reply:
Ah, we may not need to provide `match_substring_regex*`.
How about providing only `match_substring?`?

Reply:
Is it usable when we pass a string regular expression directly?

`match_substring_regex?("[dr]ow")`

Reply:
Hmm.
I don't think that it's usable.
Do you have any use case for it rather than `match_substring?(/[dr]ow/)` style?

If you don't have any use case for it, how about not providing `match_substring_regexp?` for now? If we get a request for it from an user, we can reconsider this.

Reply:
No I don't have any specific use case.
But I feel it is wasteful to 'construct Regexp => pick up source => apply it as string to match_substring_regex()`.
Anyway I will follow you. Thanks.

Reply:
OK.
Then we can improve the current APIs later.

I think that `match_substring_regexp?` isn't a good API because existing built-in Ruby objects don't provide `match_substring_regexp?` (`String` provides `start_with?`. So `start_with?` is a good API for Rubyists.) And compute function name isn't `match_substring_regexp` (`..._regex`). So `match_substring_regexp?` confuses us.

I think that we can find better names for this. For example:

* `column.match?(string_pattern)` -> `match_substring_regex(column, string_pattern)` (`String#match?` exists.)
* `column.match?(regexp)` -> `match_substring_regex(column, regexp.source)` (`String#match?` exists.)
* `column.include?(string)` -> `match_substring(column, string)` (`String#include?` exists.)

Reply:
I agree. Thanks!